tests/h/services/search_index/_queue_test.py

```diff
@@ -0,0 +1,364 @@
+            assert annotation_id not in batch_indexer.index.call_args[0][0]
+
+    def test_if_the_annotation_isnt_in_the_DB_it_deletes_the_job_from_the_queue(
+        self,
+        annotations,
+        annotation_ids,
+        caplog,
+        db_session,
+        queue,
+    ):
+        for annotation in annotations[:limit]:
+            db_session.delete(annotation)
+
+        queue.add_all(
+            annotation_ids[:limit],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Deleting 10 sync annotation jobs because their annotations have been deleted from the DB",
+            )
+        ]
+        assert db_session.query(Job).all() == []
+
+    def test_if_the_annotation_is_marked_as_deleted_in_the_DB_it_deletes_the_job_from_the_queue(
+        self,
+        annotations,
+        annotation_ids,
+        caplog,
+        db_session,
+        queue,
+    ):
+        for annotation in annotations[:limit]:
+            annotation.deleted = True
+
+        queue.add_all(
+            annotation_ids[:limit],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Deleting 10 sync annotation jobs because their annotations have been deleted from the DB",
+            )
+        ]
+        assert db_session.query(Job).all() == []
+
+    def test_if_the_annotation_is_missing_from_Elastic_it_indexes_it(
+        self,
+        annotation_ids,
+        batch_indexer,
+        caplog,
+        queue,
+    ):
+        queue.add_all(
+            annotation_ids[:limit],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Syncing 10 annotations that are missing from Elasticsearch",
+            )
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing(annotation_ids[:limit]).only()
+        )
+
+    def test_if_the_annotation_is_already_in_Elastic_it_removes_the_job_from_the_queue(
+        self,
+        annotations,
+        annotation_ids,
+        batch_indexer,
+        caplog,
+        db_session,
+        es,
+        queue,
+    ):
+        es.conn.search.return_value = {
+            "hits": {
+                "hits": [
+                    {
+                        "_id": annotation.id,
+                        "_source": {
+                            "updated": (annotation.updated).isoformat(),
+                        },
+                    }
+                    for annotation in annotations[:limit]
+                ],
+            },
+        }
```

Comment:
This should probably be changed to use the real Elasticsearch instead of mocking its return values